src/main/scala/org/apache/sysml/api/dl/CaffeLayer.scala

```diff
@@ -978,7 +1045,15 @@ class Convolution(val param:LayerParameter, val id:Int, val net:CaffeNetwork) ex
 }
 
 class DeConvolution(val param:LayerParameter, val id:Int, val net:CaffeNetwork) extends CaffeLayer with HasWeight with HasBias {
-  override def sourceFileName: String = "conv2d_transpose"
+  def isDepthWise():Boolean = {
+    if(param.getConvolutionParam.hasGroup && param.getConvolutionParam.getGroup != 1 && numChannels.toInt % param.getConvolutionParam.getGroup != 0) 
+      throw new DMLRuntimeException("The number of groups=" + param.getConvolutionParam.getGroup + " is not supported as it is not divisible by number of channels" + numChannels + ".")
+    param.getConvolutionParam.hasGroup && param.getConvolutionParam.getGroup != 1
+  }
+  def depth():String = if(isDepthWise) (numChannels.toInt / param.getConvolutionParam.getGroup).toString else throw new DMLRuntimeException("Incorrect usage of depth")
```

Comment:
Here, the name `depth` is semantically fine, since in depthwise transpose conv2d, each group of `M` input channels is mapped to a single output channel by a single filter of depth `M`.  However, I would still map it to a ConvolutionParam defaulted to 1, perhaps just reusing `depthMultiplier` from above.